### DISCUSSION TITLE
RCEDIT implementation is laughably bad and needs improvements.

### DISCUSSION THREAD

**Discussion by SilencedPerson** [godotengine/godot-proposals#4502] [Editor]
Honestly, Icon is very vital part of a program.
I do not understand why does Godot have such a massive problem with it.
Compared to other engines where users usually simply need to put in PNG file to have it converted to ICO automatically, Godot requires to include 3rd Party software which would be fine but it also doesn't work properly most of the time.
The icon can, and often times will, get corrupted by the engine's own icon, ending up with a blurry one despite having all the correct sizes,
Why is it so and how can it be remedied?

**Comment by Calinou** [comment]
Using rcedit to change the Windows icon is no longer required in 3.5 and later, as the ICO file embedded within the executable is no longer compressed: https://github.com/godotengine/godot/pull/57850

This allows you to use a hex editor to replace the icon, or any custom tool that performs hex editing behind the scenes.

**Reply by SilencedPerson** [reply]
those are actually great news.
